recipes/VideoBotsStats.py

```diff
@@ -309,14 +349,105 @@ def render(self):
                 )
                 csv = df.to_csv()
                 b64 = base64.b64encode(csv.encode()).decode()
-                gui.html(
-                    f'<a href="data:file/csv;base64,{b64}" download="{bi.name}.csv" class="btn btn-theme btn-secondary">Download CSV File</a>'
+                gui.js(
+                    dedent(
+                        # language=javscript
+                        f"""
+                        (function() {{
+                            let el = document.createElement("a");
+                            el.setAttribute("href", "data:file/csv;base64,{b64}");
+                            el.setAttribute("download", "{slugify(bi.name)}.csv");
+                            document.body.appendChild(el);
+                            el.click();
+                            document.body.removeChild(el);
+                        }})();
+                        """
+                    )
```

Comment:
this lets the server handle the effect after click - so that we don't prepare the CSV file on every page load